staging/src/k8s.io/controller-manager/controller/interfaces.go

```diff
@@ -36,3 +40,15 @@ type Debuggable interface {
 	// The handler will be accessible at "/debug/controllers/{controllerName}/".
 	DebuggingHandler() http.Handler
 }
+
+// HealthCheckable defines a controller that allows the controller manager
+// to include it in the health checks.
+//
+// If a controller implements HealthCheckable, and the returned HealthChecker
```

Comment:
very minor: These details seem to belong with the godoc for the below `HealthChecker()` func declaration.

Reply:
I think it is both. At the type level, the point is that the controller still has a chance to disable health check even if it implements this interface.

Reply:
Add that to `HealthChecker()` too